### PR TITLE
Bug fix from PED

### LINKED ISSUE #175: [PE-D][Tester B] [TP] Invalid command message sent with spacings in front of a valid command
Spacings in front of a valid command triggers the invalid command message. For the non-observant users, they may be confused, because what they have input is valid. You may consider trimming the inputs, such that spacing would not affect the validity of inputs, making it easier for users to use. 

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: glenda-1506/ped#7

### PR DESCRIPTION
**Pull request by ThienDuc3112**
Fix #175 #176

### CODE REVIEW

**Comment by ThienDuc3112** [comment]
Fix #174 as well

File: src/main/java/tutorlink/command/AddComponentCommand.java [hunk 1]
```diff
@@ -27,11 +27,10 @@ public CommandResult execute(AppState appState, HashMap<String, String> hashmap)
...
 
-        if((weightage + Component.totalWeight) > MAX_WEIGHT) {
+        int totalWeight = weightage + appState.components.getTotalWeighting();
+        if ((totalWeight) > MAX_WEIGHT) {
             throw new InvalidWeightingException(String.format(Commons.ERROR_INVALID_TOTAL_WEIGHTING,
-                    Component.totalWeight + weightage));
-        } else {
-            Component.totalWeight += weightage;
+                    totalWeight));
         }
```
Comment (RCPilot1604) [inline src/main/java/tutorlink/command/AddComponentCommand.java:34]: Computing totalWeight every time significantly reduces the code but also adds overhead. 

However for the sake of defensibility I think this is acceptable.

**Review by RCPilot1604 — APPROVED** [review]
LGTM

**Comment by RCPilot1604** [comment]
Fixes #170

**Pull request merged**